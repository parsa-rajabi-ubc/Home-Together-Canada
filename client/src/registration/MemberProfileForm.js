/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.16
 *
 * @Description: Member profile static form Component. Returns a form.
 *
 */
import React, { useState } from 'react';
import TextArea from '../common/forms/TextArea';
import RadioButton from "../common/forms/RadioButton";
import Dropdown from "../common/forms/Dropdown";
import Button from "../common/forms/Button";
import statuses from "../common/forms/Status";
import workStatuses from "../common/forms/WorkStatus";
import shareLimits from "../common/forms/ShareLimits";
import PreferredArea from "../common/forms/preferredArea";
import YNButton from "../common/forms/YNButtons";

//Returns a Form with fields
function MemberProfileForm() {
    const [gender, setGender] = useState(true);
    const [partner, setPartner] = useState(false);
    const [group,setGroup] = useState(false);
    const [petFriendly, setPetFriendly] = useState(false);
    const [smoking, setSmoking] = useState(false);
    const [mobilityIssues, setMobilityIssues] = useState(false);
    const [hasAllergies, setHasAllergies] = useState(false);
    const [religious, setReligious] = useState(false);
    const [hasDiet, setHasDiet] = useState(false);
    const [hasHome, setHasHome] = useState(false);

    return (
        <div>
            <h1>Member Sign Up:</h1>
            <hr/>
            <form>
                <div>
                    <p>Gender</p>
                    <RadioButton label="Male: " name="male" value="male" checked={gender} onChange={onchange}/>
                    <RadioButton label="Female: " name="female" value="female" checked={!gender} onChange={onchange}/>
                    <RadioButton label="Other: " name="other" value="other" checked={!gender} onChange={onchange}/>
                    <TextArea label="Preferred Pronoun: " placeholder="Pronoun"/>
                </div>
                <TextArea label="Year of Birth: " placeholder="" onChange={onchange} />
                <div>
                    <p>Status</p>
                    <Dropdown name="status" title={"Status"} items={statuses} onChange={onchange}/>
                    <TextArea label="Partner's username(s): " placeholder="username(s)" onChange={onchange}/>
                </div>
                <div>
                    <p>I am open to sharing with up to: </p>
                    <Dropdown name="shareLimit" items={shareLimits} title={"Share Limit"} onChange={onchange}/>
                </div>
                <div>
                    <p>Work Status</p>
                    <Dropdown name="work status" title={"Work Status"} items={workStatuses} onChange={onchange}/>
                    <p>Max monthly rent</p>
                    <span>$</span><input type="number" min="0.00" step="0.01" placeholder="1500.00"/>
                </div>
                <PreferredArea/>
                <Button label="Add another area: " value="+" onClick={onclick}/>
                <YNButton label="Pet Friendly?:" checked={petFriendly} onChange={onchange}/>
                <YNButton label="Smoking Friendly?:" checked={smoking} onChange={onchange}/>
                <YNButton label="Health or Mobility Issues?:" checked={mobilityIssues} onChange={onchange}/>
                <YNButton label="Allergies?:" checked={hasAllergies} onChange={onchange}/>
                <YNButton label="Religion is important to me?:" checked={religious} onChange={onchange}/>
                <YNButton label="The diet of others in my home is important to me?:" checked={hasDiet} onChange={onchange}/>
                <YNButton label="I have a home to share?:" checked={hasHome} onChange={onchange}/>

            </form>
        </div>
    );
}
export default MemberProfileForm;