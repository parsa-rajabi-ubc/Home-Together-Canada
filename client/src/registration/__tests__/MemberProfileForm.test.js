/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.19
 *
 * @Description: Member profile form Component snapshot test. Returns a snapshot.
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import MemberProfileForm from "../MemberProfileForm";
import { BrowserRouter as Router } from 'react-router-dom';

describe('MemberProfileForm', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            const MemberProfile = (
                <Router>
                    <MemberProfileForm />
                </Router>
            )
            //when
            const component = renderer.create(MemberProfile).toJSON();
            //then
            expect(component).toMatchSnapshot();
        });
    })
})