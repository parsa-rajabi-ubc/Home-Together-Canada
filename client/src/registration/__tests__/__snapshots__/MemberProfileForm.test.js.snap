// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberProfileForm Snapshot test should render correctly regardless of properties 1`] = `
<div>
  <h1>
    Member Sign Up:
  </h1>
  <hr />
  <form>
    <div>
      <p>
        Gender
      </p>
      <label>
        Male: 
        <input
          checked={false}
          name="gender"
          onChange={[Function]}
          type="radio"
          value="male"
        />
      </label>
      <label>
        Female: 
        <input
          checked={false}
          name="gender"
          onChange={[Function]}
          type="radio"
          value="female"
        />
      </label>
      <label>
        Other: 
        <input
          checked={false}
          name="gender"
          onChange={[Function]}
          type="radio"
          value="other"
        />
      </label>
    </div>
    <div>
      <span>
        Status
      </span>
      <div>
        <div>
          <div
            className=" css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-yk16xz-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-g1d714-ValueContainer"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Family Status
                </div>
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-2-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-6q0nyr-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <span>
        I am open to sharing with up to: 
      </span>
      <div>
        <div
          className=" css-2b097c-container"
          onKeyDown={[Function]}
        >
          <div
            className=" css-yk16xz-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-g1d714-ValueContainer"
            >
              <div
                className=" css-1wa3eu0-placeholder"
              >
                Share Limit
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-3-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-6q0nyr-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <p>
        Min and max monthly rent
      </p>
      <span>
        $
      </span>
      <input
        min="0.00"
        onChange={[Function]}
        placeholder="500.00"
        step="0.01"
        type="number"
        value="500.00"
      />
      <span>
        $
      </span>
      <input
        min="0.00"
        onChange={[Function]}
        placeholder="1500.00"
        step="0.01"
        type="number"
        value="1500.00"
      />
    </div>
    <span>
      Interested Area
    </span>
    <div>
      <div>
        <div
          className=" css-2b097c-container"
          onKeyDown={[Function]}
        >
          <div
            className=" css-yk16xz-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-g1d714-ValueContainer"
            >
              <div
                className=" css-1wa3eu0-placeholder"
              >
                Province
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-4-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-6q0nyr-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <label>
      <div>
        <p>
          Pet Friendly?:
        </p>
        <label>
          yes: 
          <input
            checked={true}
            name="petFriendly"
            onChange={[Function]}
            type="radio"
            value="yes"
          />
        </label>
        <label>
          no: 
          <input
            checked={false}
            name="petFriendly"
            onChange={[Function]}
            type="radio"
            value="no"
          />
        </label>
        <label>
          If Yes, Elaborate: 
          <input
            disabled={false}
            placeholder=""
            type="text"
          />
        </label>
      </div>
    </label>
    <span>
      
    </span>
    <label>
      <div>
        <p>
          Smoking Friendly?:
        </p>
        <label>
          yes: 
          <input
            checked={true}
            name="smoking"
            onChange={[Function]}
            type="radio"
            value="yes"
          />
        </label>
        <label>
          no: 
          <input
            checked={false}
            name="smoking"
            onChange={[Function]}
            type="radio"
            value="no"
          />
        </label>
        <label>
          If Yes, Elaborate: 
          <input
            disabled={false}
            placeholder=""
            type="text"
          />
        </label>
      </div>
    </label>
    <label>
      <div>
        <p>
          Health or Mobility Issues?:
        </p>
        <label>
          yes: 
          <input
            checked={true}
            name="mobile"
            onChange={[Function]}
            type="radio"
            value="yes"
          />
        </label>
        <label>
          no: 
          <input
            checked={false}
            name="mobile"
            onChange={[Function]}
            type="radio"
            value="no"
          />
        </label>
        <label>
          If Yes, Elaborate: 
          <input
            disabled={false}
            placeholder=""
            type="text"
          />
        </label>
      </div>
    </label>
    <label>
      <div>
        <p>
          Allergies?:
        </p>
        <label>
          yes: 
          <input
            checked={true}
            name="allergies"
            onChange={[Function]}
            type="radio"
            value="yes"
          />
        </label>
        <label>
          no: 
          <input
            checked={false}
            name="allergies"
            onChange={[Function]}
            type="radio"
            value="no"
          />
        </label>
        <label>
          If Yes, Elaborate: 
          <input
            disabled={false}
            placeholder=""
            type="text"
          />
        </label>
      </div>
    </label>
    <label>
      <div>
        <p>
          Religion is important to me?:
        </p>
        <label>
          yes: 
          <input
            checked={true}
            name="religion"
            onChange={[Function]}
            type="radio"
            value="yes"
          />
        </label>
        <label>
          no: 
          <input
            checked={false}
            name="religion"
            onChange={[Function]}
            type="radio"
            value="no"
          />
        </label>
        <label>
          If Yes, Elaborate: 
          <input
            disabled={false}
            placeholder=""
            type="text"
          />
        </label>
      </div>
    </label>
    <label>
      <div>
        <p>
          The diet of others in my home is important to me?:
        </p>
        <label>
          yes: 
          <input
            checked={true}
            name="diet"
            onChange={[Function]}
            type="radio"
            value="yes"
          />
        </label>
        <label>
          no: 
          <input
            checked={false}
            name="diet"
            onChange={[Function]}
            type="radio"
            value="no"
          />
        </label>
        <label>
          If Yes, Elaborate: 
          <input
            disabled={false}
            placeholder=""
            type="text"
          />
        </label>
      </div>
    </label>
    <label>
      <div>
        <p>
          I have a home to share?:
        </p>
        <label>
          yes: 
          <input
            checked={true}
            name="hasHome"
            onChange={[Function]}
            type="radio"
            value="yes"
          />
        </label>
        <label>
          no: 
          <input
            checked={false}
            name="hasHome"
            onChange={[Function]}
            type="radio"
            value="no"
          />
        </label>
        <label>
          If Yes, Elaborate: 
          <input
            disabled={false}
            placeholder=""
            type="text"
          />
        </label>
      </div>
    </label>
    <label>
      Tell others about yourself: 
      <textarea
        cols="50"
        name="aboutSelf"
        onChange={[Function]}
        placeholder="What is important to you, and why do you want to share a home?"
        rows="8"
      />
    </label>
  </form>
</div>
`;
